src/dbnode/client/config.go

```diff
@@ -177,19 +177,40 @@ func (c Configuration) NewAdminClient(
 
 	v := NewAdminOptions().
 		SetTopologyInitializer(envCfg.TopologyInitializer).
-		SetWriteConsistencyLevel(c.WriteConsistencyLevel).
-		SetReadConsistencyLevel(c.ReadConsistencyLevel).
-		SetClusterConnectConsistencyLevel(c.ConnectConsistencyLevel).
-		SetBackgroundHealthCheckFailLimit(c.BackgroundHealthCheckFailLimit).
-		SetBackgroundHealthCheckFailThrottleFactor(c.BackgroundHealthCheckFailThrottleFactor).
-		SetWriteRequestTimeout(c.WriteTimeout).
-		SetFetchRequestTimeout(c.FetchTimeout).
-		SetClusterConnectTimeout(c.ConnectTimeout).
-		SetWriteRetrier(c.WriteRetry.NewRetrier(writeRequestScope)).
-		SetFetchRetrier(c.FetchRetry.NewRetrier(fetchRequestScope)).
 		SetChannelOptions(xtchannel.NewDefaultChannelOptions()).
 		SetInstrumentOptions(iopts)
 
+	if c.WriteConsistencyLevel != nil {
+		v = v.SetWriteConsistencyLevel(*c.WriteConsistencyLevel)
+	}
+	if c.ReadConsistencyLevel != nil {
+		v = v.SetReadConsistencyLevel(*c.ReadConsistencyLevel)
+	}
+	if c.ConnectConsistencyLevel != nil {
+		v.SetClusterConnectConsistencyLevel(*c.ConnectConsistencyLevel)
+	}
+	if c.BackgroundHealthCheckFailLimit != nil {
+		v = v.SetBackgroundHealthCheckFailLimit(*c.BackgroundHealthCheckFailLimit)
+	}
+	if c.BackgroundHealthCheckFailThrottleFactor != nil {
+		v = v.SetBackgroundHealthCheckFailThrottleFactor(*c.BackgroundHealthCheckFailThrottleFactor)
+	}
+	if c.WriteTimeout != nil {
+		v = v.SetWriteRequestTimeout(*c.WriteTimeout)
```

Comment:
Small nit, but call it `*c.WriteRequestTimeout` or `v.SetWriteTimeout`

Reply:
Same for Fetch below

Reply:
these were already defined this way and I don't want to fix a bunch fo random code that depends on the name of these fields in this struct in this P.R